src/main/java/co/helmethair/scalatest/ScalatestEngine.java

```diff
@@ -29,9 +37,16 @@ public String getId() {
     public TestDescriptor discover(EngineDiscoveryRequest discoveryRequest, UniqueId uniqueId) {
         ScalatestEngineDescriptor engineDescriptor = new ScalatestEngineDescriptor(uniqueId, ID);
 
-        List<Class<? extends Suite>> classes = discoverClassSelectors(discoveryRequest);
+        return runtime.discover(engineDescriptor,
+            union(discoverClassSelectors(discoveryRequest), discoverUniqueIdSelectors(discoveryRequest)),
```

Comment:
Can we please change this to something like this below? No real difference just saves a few lines of code. 
```suggestion
            Stream.concat(
                discoverClassSelectors(discoveryRequest).stream(),
                discoverUniqueIdSelectors(discoveryRequest).stream()
            ).collect(Collectors.toSet())
```

Reply:
Nice, minor detail the returned set might be mutable `Collectors.toUnmodifiableSet()` is only available from java 10

Reply:
Thank you very much, its fine, for me the functional style matters more